### ISSUE TITLE
StackOverflowError mapping hibernate4 proxy.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#20]
```
What steps will reproduce the problem?
1. Create MapperFactory with HibernateUnenhanceStrategy
2. Load hibernate entity, and map to DTO
3. If entity is wrapped to proxy and have associations with proxy, then orika 
throw StackOverflowError

What is the expected output? What do you see instead?
DTO is expected. I have infinite recursion instead.

What version of the product are you using? On what operating system?
orika 1.1.2 and 1.1.1, windows 7, jsdk 7

More details are in attached archive.

```

Original issue reported on code.google.com by `dkhomya...@gmail.com` on 25 May 2012 at 12:27

Attachments:
- [test-src.rar](https://storage.googleapis.com/google-code-attachments/orika/issue-20/comment-0/test-src.rar)

**Comment by GoogleCodeExporter** [comment]
```
Fixed in the trunk (1.1.4-SNAPSHOT)
```

Original comment by `matt.deb...@gmail.com` on 26 May 2012 at 11:45
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `elaat...@gmail.com` on 27 May 2012 at 2:31
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.1.4 release
```

Original comment by `matt.deb...@gmail.com` on 27 May 2012 at 4:34

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot for responsiveness! Unfortunally, I've found regression in 1.1.4 
http://code.google.com/p/orika/issues/detail?id=21
```

Original comment by `dkhomya...@gmail.com` on 29 May 2012 at 8:05
